crowbar_framework/app/controllers/api/upgrade_controller.rb

```diff
@@ -57,7 +57,56 @@ def prepare
   api_version "2.0"
   def services
     if request.post?
-      head :not_implemented
+
+      # prepare the scripts for various actions in one chef run
+      service_object = CrowbarService.new(Rails.logger)
+      service_object.prepare_nodes_for_os_upgrade
+
+      # initiate the services shutdown in all clusters
+      # These are the non essential services that could be shut down everywhere
+      # FIXME: non-HA case not being handled
+      NodeObject.find("pacemaker_founder:true").each do |node|
+        node.ssh_cmd("/usr/sbin/crowbar-shutdown-services-before-upgrade.sh")
+      end
+      # FIXME: dump database on database node - currently missing
+
+      head :ok
+    else
+      render json: [], status: :not_implemented
```

Comment:
I'm not sure if it makes sense to show result of https://github.com/crowbar/crowbar-core/pull/653/files#diff-f76725e43d676e1fb0ea7adda50cfc00R28

But maybe yes... have you count with this information for the UI?